### ISSUE TITLE
[Facebook] - Opt-in send message

### ISSUE THREAD

**Issue by ribb3r** [mpociot/botman#287]
Hi,

I just added a FacebookOptinDriver in #260 for the [checkbox plugin](https://developers.facebook.com/docs/messenger-platform/plugin-reference/checkbox-plugin?locale=en_US). After receiving the callback opt-in event, I would like to send a message to the user. Unfortunately, the structure for opt-in text messages is slightly different than normal text-only message.

**normal message:**
```
{
  "recipient": {
    "id": "USER_ID"
  },
  "message": {
    "text": "hello, world!"
  }
}
```

**opt-in message:**
```
{
  "recipient": {
    "user_ref":"UNIQUE_REF_PARAM"
  }, 
  "message": {
    "text":"hello, world!"
  }
}
```

So the user is addressed via recipient.**user_ref** (provided via webhook) and not recipient.**id**

Should we create a new template for this or use existing?

Thank you.

**Comment by mpociot** [comment]
I don't quite get what you mean.

This means that we can not reply to checkbox messages directly using the user reference?

**Comment by ribb3r** [comment]
ok, it's a bit complicated :-)

Imagine the user checked the checkbox. Via the new FacebookOptinDriver from #260, we can directly handle this event with:

```
$botman->hears('/optin', function (BotMan $bot) {
    $bot->reply('Thanks for checking the checkbox');
});
```

If I am not wrong, `$bot->reply('Thanks for checking the checkbox');` will generate the following json structure:

```
{
  "recipient": {
    "id": "USER_ID"
  },
  "message": {
    "text": "Thanks for checking the checkbox"
  }
}
```

Unfortunately, this won't work for optin events, because the required structure is:

```
{
  "recipient": {
    "user_ref":"UNIQUE_REF_PARAM"
  }, 
  "message": {
    "text":"Thanks for checking the checkbox"
  }
}
```

So the json structure is different. It's recipient.user_ref and not recipient.id

**Comment by mpociot** [comment]
Ahh I see. We should then just modify the reply method in the new opt-in driver!

**Comment by ribb3r** [comment]
Ok, I'll create a pull request for this.

**Comment by ribb3r** [comment]
Fixed with #293

**Comment by islamw** [comment]
Guys, can anyone to show how to use this driver in controller?
